muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/base/material-loader/fields/connect-field.tsx

```diff
@@ -122,14 +124,22 @@ export default class ConnectField extends React.Component<ConnectFieldProps, Con
       modified: false,
       synced: true,
       syncError: null,
-      answerState: null
+      answerState: null,
+
+      fieldSavedState: null,
     }
 
     this.swapField = this.swapField.bind(this);
     this.swapCounterpart = this.swapCounterpart.bind(this);
     this.pickField = this.pickField.bind(this);
     this.cancelPreviousPick = this.cancelPreviousPick.bind(this);
     this.triggerChange = this.triggerChange.bind(this);
+    this.onFieldSavedStateChange = this.onFieldSavedStateChange.bind(this);
+  }
+  onFieldSavedStateChange(savedState: FieldStateStatus){
+    this.setState({
+      fieldSavedState: savedState
+    });
```

Comment:
Might be nitpicking but for readability consider using spacing between function blocks and even though function that is happening is simple, use js documentation for function even if it is just adding name. It also helps readability by structuring code.